pinot-tools/src/main/java/org/apache/pinot/tools/PinotDataAndQueryAnonymizer.java

```diff
@@ -0,0 +1,1332 @@
+  // as in source table segments
+  private final Map<String, OrigAndDerivedValueHolder> _origToDerivedValueGlobalDictionary;
+  // used to map the original column name to a generated column name
+  private final Map<String, String> _origToDerivedColumnsMap;
+
+  private final Stopwatch _timeToBuildDictionaries = Stopwatch.createUnstarted();
+  private final Stopwatch _timeToGenerateAvroFiles = Stopwatch.createUnstarted();
+
+  private Schema _pinotSchema = null;
+  private org.apache.avro.Schema _avroSchema = null;
+
+  private final Map<String, FieldSpec.DataType> _columnToDataTypeMap;
+  // name of columns to build global dictionary for and corresponding total cardinality
+  private final Map<String, Integer> _globalDictionaryColumns;
+  // name of time (or time derived) columns for which we will retain data
+  private final Set<String> _timeColumns;
+
+  private String[] _segmentDirectories;
+
+  /**
+   * Create an instance of PinotDataGenerator
+   * @param outputDir parent directory where avro files will be generated
+   * @param segmentDir directory containing segment
+   * @param fileNamePrefix generated avro file name prefix
+   */
+  public PinotDataAndQueryAnonymizer(
+      String segmentDir,
+      String outputDir,
+      String fileNamePrefix,
+      Map<String, Integer> globalDictionaryColumns,
+      Set<String> timeColumns) {
+    _outputDir = outputDir;
+    _segmentDir = segmentDir;
+    _filePrefix = fileNamePrefix;
+    _origToDerivedValueGlobalDictionary = new HashMap<>();
+    _origToDerivedColumnsMap = new HashMap<>();
+    _columnToDataTypeMap = new HashMap<>();
+    _globalDictionaryColumns = globalDictionaryColumns;
+    _timeColumns = timeColumns;
+
+    for (String column : timeColumns) {
+      // sometime the predicates can also be on columns which the user
+      // wants to retain the data for as is and thus these columns will be
+      // part of filter column set as well.
+      // But since the values are retained for these columns, we
+      // don't need to build global dictionary for them. So remove
+      // these columns from the filter column set.
+      _globalDictionaryColumns.remove(column);
+    }
+
+    LOGGER.info("Columns to retain data for");
+    for (String column : _timeColumns) {
+      LOGGER.info("Column name: " + column);
+    }
+
+    LOGGER.info("Columns to build global dictionary for");
+    for (Map.Entry<String, Integer> entry : _globalDictionaryColumns.entrySet()) {
+      LOGGER.info("Column name: " + entry.getKey() + " cardinality: " + entry.getValue());
+    }
+  }
+
+  /*****************************************************
+   *                                                   *
+   *             Global Dictionary Builder             *
+   *                                                   *
+   *****************************************************/
+
+  /*
+   * Global Dictionary Implementation Notes
+   *
+   * We build global dictionary for a specific set of columns
```

Comment:
You can reduce the number of lines in these comments by making lines wider?

Reply:
Done